ntoskrnl/ke/amd64/mp.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Architecture specific source file to hold multiprocessor functions
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES *****************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS *******************************************************************/
+
+#define PHYSICAL_ADDRESS  LARGE_INTEGER
+PHYSICAL_ADDRESS HighestPhysicalAddress;
+ULONG ProcessorCount = 0;
+
+struct APInfo
+{
+    KPCR pcr;
+    KTSS tss;
+    ETHREAD thread;
+    KDESCRIPTOR GdtDesc;
+    KDESCRIPTOR IdtDesc;
+} APInfo;
+
+/* FUNCTIONS *****************************************************************/
+VOID
+NTAPI
+KeStartAllProcessors()
+{
+     /* Prep some globals for the AP */
+    PVOID KernelStack;
+    KPROCESSOR_STATE ProcessorState;
+    KDESCRIPTOR GdtDesc, IdtDesc;
+
+    HighestPhysicalAddress.QuadPart = UINT64_MAX;
+    __sgdt(&GdtDesc.Limit);
+    __sidt(&IdtDesc.Limit);
+
+    /* Attempt to allocate memory for the new setup */
+    SIZE_T APInfoSize = sizeof(struct APInfo) + 2;
```

Comment:
`+ 2`? Is that your memory donation for a charitable cause?

Reply:
Im going to remember i did this and cringe at my completely ignorance.
fixed it. ill push it next time i push everything

it was a donation to the Justin is a little special foundation